### DISCUSSION TITLE
Allow to define the schema with Javascript

### DISCUSSION THREAD

**Discussion by karimpanacci** [prisma/prisma#22768] [Q&A]
Prisma is amazing, but one thing I hate is the prisma schema... it would be much more useful to have the schema written in javascript, like TypeORM does... this is because defining the schema in javascript allows you to be much more flexible, for example I can define a table by implementing a parent class so that I have some properties in common on multiple tables...

And so many other things...

**Comment by ludralph** [comment]
Hi @karim2012 👋 

Thank you for raising this question.

Prisma's approach to data modeling is indeed different from some other ORMs like TypeORM. Prisma uses a declarative Prisma schema file to define the data model, which is a central concept in Prisma's design philosophy.  Prisma schema provides a clear and human-readable representation of your data model and serves as the single source of truth for your database schema.

While you prefer the flexibility of defining schemas in JavaScript, Prisma's schema has its own set of advantages. For instance, it allows for a clean separation of concerns by decoupling business logic from the shape of data, and it enables powerful tooling like the Prisma Visual Studio Code extension, which offers syntax highlighting, autocompletion, and quick fixes to make data modeling more efficient.

Moreover, the Prisma schema is designed to be self-explanatory and expressive, allowing you to effectively design your desired database schema without falling back to raw SQL for relational databases. It also provides a tight feedback loop with automated tooling that informs you about errors in your schema and suggests ways to avoid common pitfalls in database schema design.

While Prisma's approach might not offer the same kind of flexibility in terms of JavaScript inheritance and shared properties across tables that you're looking for, it does provide a robust and productive environment for many developers, especially in collaborative team settings.

You should also note that we are constantly adding new features to Prisma ORM and the support for inheritance has been requested for in this [issue](https://github.com/prisma/prisma/issues/12604).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
